genesis/auth.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+
+import certifi
+
+
+class SASLAuth(object):
+    """Attach SASL-based authentication to a client.
+
+    Returns client-based auth options when called.
+
+    Parameters
+    ----------
+    user : `str`
+        Username to authenticate with.
+    password : `str`
+        Password to authenticate with.
+    ssl : `bool`, optional
+        Whether to enable SSL (enabled by default).
+    method : `str`, optional
+        The SASL method to authenticate, default = PLAIN.
+        Options are [PLAIN, SCRAM-SHA-256, SCRAM-SHA-512].
+    ssl_ca_location : `str`, optional
+        If using SSL via a self-signed cert, a path/location
+        to the certificate.
+
+    """
+
+    _methods = ["PLAIN", "SCRAM-SHA-256", "SCRAM-SHA-512"]
```

Comment:
This seems like a good opportunity for an [Enum](https://docs.python.org/3/library/enum.html). You would gain a spot to document each option, could ditch the `assert`, and would help users avoid dumb typos.

Reply:
An Enum sounds reasonable, how do you see the implementation being? As in, should the user still pass in a string and the `SASLAuth` class handle it internally or have the user pass in the enum directly?

One other complication is that `SCRAM-SHA-256` wouldn't be a valid variable, so we would need to map those to something else, and wouldn't end up being one-to-one with the sasl.mechanisms options in librdkafka.

Reply:
I was picturing the enum being used directly by users, something like this:
```python
class AuthMethod(enum.Enum): 
     PLAIN = 1 
     SCRAM_SHA_256 = 2 
     SCRAM_SHA_512 = 3 

     def __str__(self): return self.name.replace("_", "-")
```

Users use it like 
  
```python
auth = SASLAuth("u", "pw", method=AuthMethod.SCRAM_SHA_256)
```

I don't know whether `__str__` is sufficient for `certifi` - you might need to explicitly do `self._config["sasl.mechanism"] = str(method)`.

Reply:
Sounds good to me, thanks for the clarification. I've added this in the latest commit, but with `SASLMethod` instead.

Reply:
I like it!